lib/daru/index.rb

```diff
@@ -51,8 +51,8 @@ def map(&block)
 
     def initialize index
       index = 0                         if index.nil?
-      index = Array.new(index) { |i| i} if index.is_a? Integer
-      index = index.to_a                if index.is_a? Daru::Index
+      index = index.times.to_a          if index.is_a? Integer
+      index = index.to_a                if index.is_a? Enumerable
 
       @relation_hash = {}
       index.each_with_index do |n, idx|
```

Comment:
BTW, as far as I can see, the code of this each_with_index could be replaces with just `@relations_hash = index.each_with_index.to_h`